File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    let set = new Set();
    
    for (let size = 1; size <= elements.length; size++) {
        for (let start = 0; start < elements.length; start++) {
            sum = 0;
            for (let i = start; i < start + size; i++) {
                sum += elements[i % elements.length];
            }
            set.add(sum)
        }
    }
    
    return set.size;
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        
        Set<Integer> set = new HashSet<>();
        for (int size = 1; size < elements.length + 1; size++) {
            for (int start = 0; start < elements.length; start++) {
                int sum = 0;
                for (int i = start; i < start + size; i++) {
                    sum += elements[i % elements.length];
                }
                set.add(sum);
            }
        }
        
        return set.size();
    }
}